apps/evm_family/base/base_app.c

```diff
@@ -0,0 +1,154 @@
+ * limitation fees for hosting or consulting/ support services related
+ * to the Software), a product or service whose value derives, entirely
+ * or substantially, from the functionality of the Software. Any license
+ * notice or attribution required by the License must also include
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+
+#include "base_app.h"
+
+#include <stddef.h>
+
+#include "evm_main.h"
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/**
+ * @brief Whitelisted contracts with respective token symbol
+ * @details A map of Ethereum contract addresses with their token symbols. These
+ * will enable the device to verify the ERC20 token transaction in a
+ * user-friendly manner.
+ *
+ * @see erc20_contracts_t
+ */
+extern const erc20_contracts_t base_contracts[];
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/**
+ * @brief Checks if the provided token address is whitelisted and return the
+ * matching contract instance.
+ *
+ * @param address Reference to the buffer containing the token address
+ * @param contract Pointer to store the matched contract address instance
+ *
+ * @return bool Indicating if the provided token address is whitelisted
+ * @return true If the address matches to an entry in the whitelist
+ * @return false If the address does not match to an entry in the whitelist
+ */
+static bool is_token_whitelisted(const uint8_t *address,
+                                 const erc20_contracts_t **contract);
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+
+static const evm_config_t base_app_config = {
+    .lunit_name = "ETH",
+    .name = "BASE",
+    .chain_id = 8453,
+
+    // whitelisted contracts
+    .is_token_whitelisted = is_token_whitelisted,
+};
+
+static const cy_app_desc_t base_app_desc = {
+    .id = 18,
+    .version =
```

Comment:
change branch branch to merge "develop"
change id to 19.

Reply:
also resolve the commit checks in CI

Reply:
corresponding proto files needs to be added
https://github.com/Cypherock/cypherock-common